### PR TITLE
Feature: return validated attributes of JWT

### PR DESCRIPTION
**Pull request by martinjaimem**
**This PR adds the possibility to see the decoded JWT data in the sign-in/up.** 
This is needed when integrating with devise, as there are attributes such as email that we need to retrieve from the JWT.

As the JWT needs to be decoded in order to be validated, then I thought that the function that validates the integrity of the JWT could also return the validated data (if valid, if not it would raise an error). If both were separated methods (lets say we have a new one that is named `decode`) then I couldn't think of a case where we would have to decode the data without validating it.

WDYT?

### CODE REVIEW

**Comment by vitogit** [comment]
I think is a good idea to return the values

File: lib/apple_sign_in/user_identity.rb [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 module AppleSignIn
   class UserIdentity
+    class JWTValidationError < StandardError; end
```
Comment (fedeagripa) [inline lib/apple_sign_in/user_identity.rb:5]: can we add some info to this custom error?
Like `fields`, `description` or something that let the dev know more about the validation error perse

File: spec/helpers/jwt_conditions_spec.rb [hunk 2]
```diff
@@ -33,65 +33,79 @@
...
+          )
         end
       end
 
       context 'when exp is not a integer' do
         let(:jwt_iat) { Time.now }
 
-        it 'returns false' do
-          expect(jwt_conditions_helper).to_not be_valid
+        it 'raise an exception' do
```
Comment (martinjaimem) [inline spec/helpers/jwt_conditions_spec.rb:54]:
```suggestion
        it 'raises an exception' do
```

File: spec/user_identity_spec.rb [hunk 3]
```diff
@@ -70,7 +69,9 @@
       let(:uid) { '1234.5678.911' }
 
       it 'returns false' do
```
Comment (martinjaimem) [inline spec/user_identity_spec.rb:71]: update this 😄

File: README.md [hunk 4]
```diff
@@ -26,7 +26,22 @@ Or install it yourself as:
...
+valid_jwt_token = 'eyJraWQiOiI4NkQ4OEtmIiwiYWxnIjoiUlMyNTYifQ.eyJpc...'
+AppleSignIn::UserIdentity.new(user_id, valid_jwt_token)
+>>  { exp: 1595279622, email: "user@example.com", email_verified: true , ...}
+
+# with an invalid JWT
+invalid_jwt_token = 'eyJraWQiOiI4NkQsd4OEtmIiwiYWxnIjoiUlMyNTYifQ.edsyJpc...'
+AppleSignIn::UserIdentity.new(user_id, invalid_jwt_token)
+>> Traceback (most recent call last):..
+>> ...
+>>  AppleSignIn::UserIdentity::JWTValidationError
```
Comment (martinjaimem) [inline README.md:43]: maybe we should update this to match: `AppleSignIn::Conditions::JWTValidationError`

File: lib/apple_sign_in/helpers/conditions/aud_condition.rb [hunk 5]
```diff
@@ -7,8 +7,10 @@ def initialize(jwt)
         @aud = jwt['aud']
       end
 
-      def valid?
-        @aud == AppleSignIn.config.apple_client_id
+      def validate!
+        return true if @aud == AppleSignIn.config.apple_client_id
+
+        raise JWTValidationError, 'jwt_aud is different to apple_client_id'
```
Comment (TimoPeraza) [inline lib/apple_sign_in/helpers/conditions/aud_condition.rb:13]: WDYT to use I18N here? just adding a config/locales folder, with en.yml file 😄

Comment (fedeagripa) [reply lib/apple_sign_in/helpers/conditions/aud_condition.rb:13]: what about doing this in a separate PR? just to don't make prs last forever

Comment (vitogit) [reply lib/apple_sign_in/helpers/conditions/aud_condition.rb:13]: That will add a dependency on the gem, I think this kind of error messages should be just for logging and the developer, not to show to the user so I don't think i18n would add value. To the user the developer should show a generic message, to not gave enough clues to possible bad actors.

**Pull request merged**